app/scripts/directives/notifications/notificationDrawerWrapper.js

```diff
@@ -0,0 +1,288 @@
+'use strict';
+
+angular
+  .module('openshiftConsole')
+  // shim for communicationg with pfNotificationDrawer
+  .component('notificationDrawerWrapper', {
+    templateUrl: 'views/directives/notifications/notification-drawer-wrapper.html',
+    controller: [
+      '$filter',
+      '$interval',
+      '$timeout',
+      '$routeParams',
+      '$rootScope',
+      'Constants',
+      'DataService',
+      'NotificationsService',
+      'EventsService',
+      function(
+        $filter,
+        $interval,
+        $timeout,
+        $routeParams,
+        $rootScope,
+        Constants,
+        DataService,
+        NotificationsService,
+        EventsService) {
+
+
+        // kill switch if watching events is too expensive
+        var ENABLE_EVENT_WATCH = _.get(Constants, 'FEATURE_FLAGS.global_event_watch_for_notification_drawer');
+        var drawer = this;
+
+        // global event watches
+        var rootScopeWatches = [];
+        // this one is treated separately from the rootScopeWatches as
+        // it may need to be updated outside of the lifecycle of init/destroy
+        var notificationListener;
+
+        var eventsWatcher;
+        var eventsFromNotifications = [];
+
+        // TODO:
+        // include both Notifications & Events,
+        // rather than destroying the map each time maintain it & add new items
+        var notificationGroupsMap = {};
+        var notificationGroups = [];
+
+        var projects = {};
+
+        var listProjects = function() {
```

Comment:
Yup, will do.  Oddly both `get` and `list` return a list, so I figured we had the `list` cached by the project picker anyway, but better to not assume.  I'll update.